### PR TITLE
Graduate MinDomains in Pod Topology Spread to beta

### PR DESCRIPTION
**Pull request by sanposhiho**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

Graduate MinDomains in Pod Topology Spread to beta

#### Which issue(s) this PR fixes:

Part of https://github.com/kubernetes/enhancements/issues/3022

#### Does this PR introduce a user-facing change?

```release-note
the `minDomains` field in Pod Topology Spread is graduated to beta
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
- [KEP]: https://github.com/kubernetes/enhancements/issues/3022
```

### CODE REVIEW

**Comment by sanposhiho** [comment]
@sftim 👍 
Updated the release note.

**Comment by sanposhiho** [comment]
/assign @liggitt
/cc @alculquicondor 

@liggitt 
Do we currently have a problem with `pull-kubernetes-verify` or `hack/update-openapi-spec.sh`..?
I've run `hack/update-openapi-spec.sh` in this branch, but `pull-kubernetes-verify` still says `Please run hack/update-openapi-spec.sh`. (This branch is based on the latest master.)

IIUC, `pull-kubernetes-verify` confirms there is no diff between the result of `hack/update-openapi-spec.sh` and the code. But, in latest master, running `hack/update-openapi-spec.sh` makes the following changes under api/open-spec/v3/. 
https://github.com/sanposhiho/kubernetes/pull/6

**Comment by liggitt** [comment]
> Do we currently have a problem with pull-kubernetes-verify or hack/update-openapi-spec.sh..?

no… other PRs are passing…

make sure you are synced with latest master and have pulled tags. `git describe` should show something starting with 1.25...

**Comment by sanposhiho** [comment]
> have pulled tags. 

Okay. I didn't know I needed to pull tags to generate valid files. Let me check.

**Comment by sanposhiho** [comment]
Let me `/hold` this, since we need to merge the enhancement PR before this PR. (In the case we cannot merge the enhancement PR before the deadline, we shouldn't merge this PR.)
https://github.com/kubernetes/enhancements/pull/3338

/hold

**Comment by sanposhiho** [comment]
/unhold

The PR for KEP updating was merged.

@liggitt @alculquicondor Could you please take a look?

File: pkg/apis/core/types.go [hunk 1]
```diff
@@ -5711,7 +5711,7 @@ type TopologySpreadConstraint struct {
 	// because computed skew will be 3(3 - 0) if new Pod is scheduled to any of the three zones,
 	// it will violate MaxSkew.
 	//
-	// This is an alpha field and requires enabling MinDomainsInPodTopologySpread feature gate.
+	// This is an beta field and requires enabling MinDomainsInPodTopologySpread feature gate.
```
Comment (alculquicondor) [inline pkg/apis/core/types.go:5714]: You need to update the field in the v1 package as well.

Comment (sanposhiho) [reply pkg/apis/core/types.go:5714]: Updated 🙏

**Pull request merged**